### DISCUSSION TITLE
What is the difference between user and end-user?

### DISCUSSION THREAD

**Discussion by miesgre** [BerriAI/litellm#3798] [Q&A] [answered]
What is the difference between user and end-user?
Is the relationship between teams, users, keys explained anywhere?

**Comment by ishaan-jaff** [comment]
User: LiteLLM exposes a /user/new endpoint to create budgets, rate limits etc for this. These are your internal users, people in companies make Users for internal employees


End User: `user passed to /chat/completions` . Think of these as customers of your product / end user 


@miesgre did I answer your question ? open to feedback

**Reply by miesgre** [reply]
I'm sorry, but I still haven't understood it.

I'm going to try to explain what we want to do to see what the best solution is.
To put ourselves in context, I work at a university and we have several use cases.


- We have some internal management tools/applications that we want to expand and make use of AI.
I understand that I could create a key for each of these applications. I want to be able to control the spending of these keys and set limits.
Should I create a user and a key for each of the applications?


- We have certain courses that teachers want to have access to a key for students to practice. We should be able to generate a key for each student enrolled in the subject.
We would like to be able to establish limits and control the spending of each student and the spending of the subject as a whole.
Should I create a team per subject? and create x keys in that team? Should I create a user for each student and assign it to the team?

In general, can a user belong to more than one team? And if it belongs to two teams, is that user's expense computed in team1 or team2? in both?

I don't understand the relationship between themes/ users/ end-users/ budgets and I can't find if it is explained anywhere, to see what is the best way to organize what you want to set up.

Thanks in advance.

**Reply by ishaan-jaff** [reply, marked as answer]
@miesgre can we hop on a call, I want to help you get to prod ASAP https://calendly.com/d/4mp-gd3-k5k/litellm-1-1-onboarding-chat

## For internal apps / tools: 
Create Team for "APP 1", Create a Key under the Team 


## For Teachers / students: 
1. Create Team: "Course 112" `/team/new` 
2. Create a "User" - "Student2, email=student2@unviersity.com" `/user/new`
3. Add Student2 to Team "course 112" using `/team/member_add'` 
4. Create a Key for Student2 `/key/generate` 
Doc for using `/team/member_add` : https://docs.litellm.ai/docs/proxy/users 
<img width="755" alt="Screenshot 2024-05-27 at 7 04 18 PM" src="https://github.com/BerriAI/litellm/assets/29436595/14d9c23c-fb92-45ab-82c0-6fddd5f3fbe1">

> can a user belong to more than one team? And if it belongs to two teams, is that user's expense computed in team1 or team2? in both?


Yes ! A student can belong to multiple teams
